src/pages/Login/KakaoRedirectHandler.js

```diff
@@ -0,0 +1,53 @@
+import React, { useEffect } from 'react';
+import { useNavigate } from 'react-router-dom';
+import { useLocation } from 'react-router-dom';
+import API from '../../config';
+
+const KakaoRedirectHandler = () => {
+  const navigate = useNavigate();
+  const location = useLocation();
+
+  const goToMain = () => {
+    navigate('/mypage');
+  };
+
+  useEffect(() => {
+    const code = location.search.substring(6);
+
+    fetch('https://kauth.kakao.com/oauth/token', {
+      method: 'POST',
+      headers: {
+        'Content-Type': 'application/x-www-form-urlencoded;',
+      },
+      body: `grant_type=authorization_code&client_id=${process.env.REACT_APP_CLIENT_ID}&code=${code}&redirect_uri=${process.env.REACT_APP_REDIRECT_URI}`,
+    })
+      .then(res => res.json())
+      .then(res => {
+        if (res.access_token) {
+          sendToken(res.access_token);
+        } else {
+          alert('다시 시도해주세요!');
+        }
+      });
+
+    const sendToken = access_token => {
+      fetch(`${API.Login}`, {
+        method: 'GET',
+        headers: {
+          Authorization: `Bearer ${access_token}`,
+        },
+      })
+        .then(res => res.json())
+        .then(res => {
+          localStorage.setItem('access_token', access_token);
+          localStorage.setItem('Authorization', res.new_token);
+
+          goToMain();
```

Comment:
```suggestion
          localStorage.setItem('access_token', access_token);
          localStorage.setItem('Authorization', res.new_token);
          navigate('/mypage')
```
* 이렇게 navigate 호출 하면 됩니다 :-)

Reply:
와 대박.. 따로 함수 생성 안해줘도 되는군요,,,,,, 감사합니다!